src/WinRT.Runtime/Marshalers.cs

```diff
@@ -157,15 +207,12 @@ public static unsafe MarshalerArray CreateMarshalerArray(string[] array)
                     m._marshalers[i] = MarshalString.CreateMarshaler(array[i]);
                     elements[i] = MarshalString.GetAbi(m._marshalers[i]);
                 };
-                success = true;
                 return m;
             }
-            finally
+            catch (Exception) when (Dispose(m))

```

Comment:
I have mentioned it in other PR. Using filters (ie `when` in C#) for cleanup is very atypical. The existing pattern with `finally` is better.

Reply:
Thanks Jan - I measured the two approaches, and finally adds 50% overhead, since it always checks a success boolean, whereas the when clause only executes in the presence of an exception.

Reply:
What is the benchmark that showed the 50% overhead? I assume that it was artificial micro-benchmark that did nothing inside the try. Once you do some actual work inside the try, the overhead should be neglible,

Using filters makes diagnostic worse. The exceptions throw inside filters are ignored, so you won't notice if anything goes wrong during Dispose.

FWIW, we are using the `bool` pattern in number of places in dotnet/runtime and we would not consider switching it to filters.

Reply:
yes, micro-benchmark with trivial body.  ok, I'll revert